include/libtorrent/http_connection.hpp

```diff
@@ -102,7 +103,7 @@ struct TORRENT_EXTRA_EXPORT http_connection
 	void get(std::string const& url, time_duration timeout = seconds(30)
 		, int prio = 0, aux::proxy_settings const* ps = 0, int handle_redirects = 5
 		, std::string const& user_agent = std::string()
-		, address const& bind_addr = address_v4::any()
+		, boost::optional<address> const& bind_addr = boost::optional<address>()
```

Comment:
do you prefer sentinel values? my experience is that sentinel values encourages half-constructed objects and possibly two-phase initialization and then it's just downhill from there :)